src/js/utils/background.js

```diff
@@ -138,14 +138,39 @@ export const backgroundStyle = (backgroundArg, theme, textColorArg) => {
   );
 
   if (background.image) {
-    // allow both background color and image, in case the image doesn't fill
-    return css`
+    const backgroundStyles = `
+      ${backgroundColor ? `background-color: ${backgroundColor};` : ''}
       background-image: ${background.image};
       background-repeat: ${background.repeat || 'no-repeat'};
       background-position: ${background.position || 'center center'};
       background-size: ${background.size || 'cover'};
-      ${backgroundColor ? `background-color: ${backgroundColor};` : ''}
+    `;
+
+    // allow both background color and image, in case the image doesn't fill
+    // when image and opacity are used together, we need to use pseudo :before
+    // to ensure that only image and background color are affected by opacity
+    // but not the container contents
+    return css`
       ${textColor ? `color: ${textColor};` : ''}
+      ${!background.opacity
+        ? backgroundStyles
+        : `position: relative;
+        z-index: 0;
```

Comment:
If the caller sets the `z-index` of the container to '-2' will this still gonna work?

Reply:
Good comment, let me test that out and confirm.

Reply:
With psuedo elements, when we create the z-index with absolute positioning, it puts it relative to the container. So in this case the image is stacked underneath the div content. If the caller sets the `z-index` of the container to `-2`, the Box itself may not be visible on the screen because it's falling underneath the general grommet wrapper, but the relative stacking between the box content and the image holds.

Reference: https://stackoverflow.com/questions/3032856/is-it-possible-to-set-the-stacking-order-of-pseudo-elements-below-their-parent-e